plutus-contract/test/Spec/TxConstraints/MustPayToPubKeyAddress.hs

```diff
@@ -0,0 +1,265 @@
+        , phase2FailureWhenUsingUnexpectedDatum
+        , phase2FailureWhenUsingUnexpectedValue
+         ]
+
+someDatum :: Datum
+someDatum = Datum $ PlutusTx.dataToBuiltinData $ PlutusTx.toData ("datum" :: P.BuiltinByteString)
+
+otherDatum :: Datum
+otherDatum = Datum $ PlutusTx.dataToBuiltinData $ PlutusTx.toData ("other datum" :: P.BuiltinByteString)
+
+adaValue :: Value.Value
+adaValue = Ada.lovelaceValueOf 5_000_000
+
+tknValue :: Value.Value
+tknValue = Value.singleton mustPayToPubKeyAddressPolicyCurrencySymbol "mint-me" 1
+
+otherTknValue :: Value.Value
+otherTknValue = Value.singleton mustPayToPubKeyAddressPolicyCurrencySymbol "dont-mint-me" 1
+
+w1PaymentPubKeyHash :: Ledger.PaymentPubKeyHash
+w1PaymentPubKeyHash = mockWalletPaymentPubKeyHash w1
+
+w2PaymentPubKeyHash :: Ledger.PaymentPubKeyHash
+w2PaymentPubKeyHash = mockWalletPaymentPubKeyHash w2
+
+w1StakePubKeyHash :: Ledger.StakePubKeyHash
+w1StakePubKeyHash = fromJust $ stakePubKeyHash $ walletToMockWallet' w1
+
+w2StakePubKeyHash :: Ledger.StakePubKeyHash
+w2StakePubKeyHash = fromJust $ stakePubKeyHash $ walletToMockWallet' w2
+
+asRedeemer :: PlutusTx.ToData a => a -> Redeemer
+asRedeemer a = Redeemer $ PlutusTx.dataToBuiltinData $ PlutusTx.toData a
+
+trace :: Contract () Empty ContractError () -> Trace.EmulatorTrace ()
+trace contract = do
+    void $ Trace.activateContractWallet w1 contract
+    void $ Trace.waitNSlots 1
+
+-- | Valid scenario using offchain and onchain constraint mustPayToPubKey with exact token value
+successfulUseOfMustPayToPubKeyWithExactTokenValue :: TestTree
+successfulUseOfMustPayToPubKeyWithExactTokenValue =
+    let options = defaultCheckOptions & changeInitialWalletValue w1 (<> tknValue)
+        onChainConstraint = asRedeemer $ MustPayToPubKey w1PaymentPubKeyHash tknValue
+        contract = do
+            let lookups1 = Constraints.plutusV1MintingPolicy mustPayToPubKeyAddressPolicy
+                tx1 = Constraints.mustPayToPubKey w1PaymentPubKeyHash adaValue
+                   <> Constraints.mustMintValueWithRedeemer onChainConstraint tknValue
+            ledgerTx1 <- submitTxConstraintsWith @UnitTest lookups1 tx1
+            awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
+
+    in checkPredicateOptions options
+    "Successful use of offchain and onchain mustPayToPubKey constraint for native token value"
+    (assertValidatedTransactionCount 1)
+    (void $ trace contract)
+
+-- | Valid scenario where the onchain mustPayToPubKey constraint expects less ada than the actual value
+successfulUseOfMustPayToPubKeyExpectingALowerAdaValue :: TestTree
+successfulUseOfMustPayToPubKeyExpectingALowerAdaValue =
+    let onChainConstraint = asRedeemer $ MustPayToPubKey w1PaymentPubKeyHash adaValue
+        contract = do
+            let lookups1 = Constraints.plutusV1MintingPolicy mustPayToPubKeyAddressPolicy
+                tx1 = Constraints.mustPayToPubKey w1PaymentPubKeyHash (Value.scale 2 adaValue)
```

Comment:
How about simply `adaValue + 1 lovelace` in order to actually test the bounds on the on-chain condition

Reply:
I'll reduce the onchain expectation by `1`. Although, it won't actually make a difference because I noticed that using both `mustPayToPubKey` and `mustMintValueWithRedeemer` produce two outputs even though the constraints could be satisfied with one output. This makes the total ada value much higher.

Reply:
It's not the constraints causing 2 outputs it's just change being returned to own wallet. I'll test by sending to other wallet.